### PR TITLE
feat(account-lib): initial setup

### PR DESCRIPTION
**Pull request by gibrangul95**
@argjv @fdmirza

### CODE REVIEW

**Review by jrafael-simtlix — DISMISSED** [review]

File: modules/account-lib/src/coin/dot/keyPair.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Secp256k1ExtendedKeyPair } from '../baseCoin/secp256k1ExtendedKeyPair';
+import { KeyPairOptions } from '../baseCoin/iface';
+
+export class KeyPair extends Secp256k1ExtendedKeyPair {
```
Comment (argjv) [inline modules/account-lib/src/coin/dot/keyPair.ts:4]: Isn't DOT ed25519?

**Review by argjv — APPROVED** [review]

**Review by tylerlevine — DISMISSED** [review]
1) we can't currently accept PRs from forks directly. Please commit to a branch inside this repository if possible
2) The commit message is lacking in information - *what* is getting the initial setup? Please leave more info for posterity along with a ticket number

File: modules/account-lib/src/coin/dot/utils.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { BaseUtils } from '../baseCoin';
+import { NotImplementedError } from '../baseCoin/errors';
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/dot/utils.ts:2]: If we are going to encode and decode address with keyring, please add unimplemented methods for encode and decode.
https://polkadot.js.org/docs/keyring/start/ss58#how-is-the-encoding-done

File: modules/account-lib/src/coin/dot/utils.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (rliberal-simtlix) [inline modules/account-lib/src/coin/dot/utils.ts:3]: Should we derive accounts with the platform? @argjv 
https://polkadot.js.org/docs/ui-keyring/start/accounts/#non-mnemonic-seeds-and-derivation

**Review by rliberal-simtlix — COMMENTED** [review]
Added my review, but many funcionality on my comment was implemented on:
https://github.com/BitGo/BitGoJS/pull/1632/files
So we need here to validate what funcionality is left.
Please mark comment as resolved for which funcionality is implemented.

File: modules/account-lib/src/coin/dot/utils.ts (hunk 3 above)
Comment (argjv) [reply modules/account-lib/src/coin/dot/utils.ts:3]: Derivation should be done in the KeyPair, but we don't really need it for DOT

**Pull request merged**